src/hotspot/share/runtime/arguments.hpp

```diff
@@ -269,6 +269,27 @@ class Arguments : AllStatic {
     ExternalProperty
   };
 
+  // RAII style wrapper representing preprocessed arguments returned from Arguments::preprocess and
+  // accepted by Arguments::parse.
+  class Preprocessed final : public StackObj {
+   private:
+    friend class Arguments;
+
+    void* _impl = nullptr;
```

Comment:
Is the `void*` typing just to avoid a forward declaration of the `PreprocessedArguments` type?

Reply:
Its to hide the implementation, so we do not have to pollute the header file. Instead of using void* I switched to using `PreprocessedImpl`.